panel/models/event-to-object.ts

```diff
@@ -25,6 +25,11 @@ SOFTWARE.
 export function serializeEvent(event: any): any {
   const data: any = {};
 
+  // support for CustomEvents: the whole `detail` object is serialized
+  if (Object.getPrototypeOf(event).initCustomEvent !== undefined) {
+    Object.assign(data, { detail: event.detail } );
```

Comment:
Maybe simply add a check if `detail` is undefined to the conditional.

Reply:
@philippjfr to resume: do you want to remove the whole poor man typechecking and just test for a `detail` attribute?

Reply:
Probably simpler in the end tbh.

Reply:
Done!

Reply:
Okay, and this does what you need?

Reply:
Yes, with https://github.com/holoviz/panel/pull/3709/commits/2ec39f04d857c98ca4a3dc26d4837dbf6c787325 I'm still able pass through the server serialized data using `.detail` as defined by MDN.

There's still an open issue about what to do if `detail` contains stuff that can't be converted to JSON.